### DISCUSSION TITLE
Is this the correct way to simulate joint friction and joint damping

### DISCUSSION THREAD

**Discussion by lyfkyle** [bulletphysics/bullet3#4356] [Q&A]
Hi, I have walked though the document and several related post. Just want to confirm here the correct way of simulating joint friction and damping in Pybullet since I want to perform torque control on my mobile robot.

So for joint friction, the value specified in URDF is NOT used and it has to be simulated through a velocity motor:
`p.setJointMotorControl2(robot_handle, joint_idx, controlMode=p.VELOCITY_CONTROL, targetVelocity=0, force=joint_friction)`

For joint damping, the value specified in URDF is applied automatically to the dynamics of the link. However, do I need to manually set the linear and angular damping field to zero?
`p.changeDynamics(robot_handle, joint_idx, linearDamping=0, angularDamping=0, jointDamping=joint_damping)`

Having accurate physics is highly desired in my case. Thanks a lot for help in advance.

**Comment by chinahuangyong** [comment]
hello, did the problem be solved?
